lib/galaxy/datatypes/text.py

```diff
@@ -1093,6 +1093,203 @@ def _looks_like_yaml(self, file_prefix: FilePrefix):
             return False
 
 
+@build_sniff_from_prefix
+class BCSLmodel(Text):
+    """BioChemical Space Language model file"""
+
+    file_ext = "bcsl.model"
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
+        """
+        Determines whether the file is in .bcsl.model format
+        """
+        content = open(file_prefix.filename, "r").read()
+        keywords = ["#! rules", "#! inits", "#! definitions"]
+        return any(keyword in content for keyword in keywords)
+
+
+@build_sniff_from_prefix
+class BCSLts(Json):
+    """BioChemical Space Language transition system file"""
+
+    file_ext = "bcsl.ts"
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
+        """
+        Determines whether the file is in .bcsl.ts format
+        """
+        is_bcsl_ts = False
+        if self._looks_like_json(file_prefix):
+            is_bcsl_ts = self._looks_like_bcsl_ts(file_prefix)
+        return is_bcsl_ts
+
+    def set_peek(self, dataset):
+        if not dataset.dataset.purged:
+            lines = "States: {}\nTransitions: {}\nUnique agents: {}\nInitial state: {}"
+            ts = json.load(open(dataset.file_name, "r"))
+            dataset.peek = lines.format(len(ts["nodes"]), len(ts["edges"]), len(ts["ordering"]), ts["initial"])
+            dataset.blurb = nice_size(dataset.get_size())
+        else:
+            dataset.peek = "file does not exist"
+            dataset.blurb = "file purged from disk"
+
+    def _looks_like_bcsl_ts(self, file_prefix: FilePrefix):
+        content = open(file_prefix.filename, "r").read()
+        keywords = ['"edges":', '"nodes":', '"ordering":', '"initial":']
+        if all(keyword in content for keyword in keywords):
+            try:
+                json.load(open(file_prefix.filename, "r"))
```

Comment:
We definitely can't load the entire file into memory here.

Reply:
The same approach is used by the base `Json` class, but yeah, it employs some additional checks for file size, so I reused its functionality in [6e6d7d6](https://github.com/galaxyproject/galaxy/pull/14362/commits/6e6d7d6ccb6ff6448ed0956962669c7bdb069c6a).